Forms/MainForm.cs

```diff
@@ -1063,6 +1296,34 @@ private void BLoadPatch_Click(object sender, EventArgs e)
             tPatch.Text = _settings.InputPatchFilename;
         }
 
+        private void bLoadPreset_Click(object sender, EventArgs e)
+        {
+            if (openPreset.ShowDialog() == DialogResult.OK)
+            {
+                _settings.UserPresetFileName = openPreset.FileName;
+                ReadPreset(_settings.UserPresetFileName);
+            }
+        }
+
+        private void bSavePreset_Click(object sender, EventArgs e)
+        {
+            if (_settings.LogicMode != LogicMode.Preset)
+            {
+                if (_settings.LogicMode == LogicMode.UserLogic && ValidateLogicFile())
```

Comment:
Why does saving settings only work in UserLogic mode?

Reply:
Yeah I'm not sure what I was trying to do here. But I fixed this.